packages/mask/src/plugins/Referral/SNSAdaptor/CreatedFarms.tsx

```diff
@@ -0,0 +1,261 @@
+        farms.push({ totalFarmRewards: farmTotalDepositMap.get(farm.farmHash) || 0, ...farm })
+    })
+
+    return farms
+}
+
+type TCreatedFarmsContent = {
+    currentChainId: ChainId
+    farms: Farm[]
+    loading: boolean
+    allTokensMap: Map<string, ERC20TokenDetailed>
+    onAdjustRewardButtonClick: (farm: Farm) => void
+}
+function CreatedFarmsContent({
+    currentChainId,
+    farms,
+    loading,
+    onAdjustRewardButtonClick,
+    allTokensMap,
+}: TCreatedFarmsContent) {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+
+    if (loading) return <CircularProgress size={50} />
+
+    if (!farms.length) return <Typography className={classes.noFarm}>{t('plugin_referral_no_created_farm')}</Typography>
+
+    return (
+        <>
+            {farms.map((farm) => (
+                <AccordionFarm
+                    key={farm.farmHash}
+                    farm={farm}
+                    allTokensMap={allTokensMap}
+                    totalValue={Number.parseFloat(farm?.totalFarmRewards?.toFixed(5) ?? '0')}
+                    accordionDetails={
+                        <Box display="flex" justifyContent="flex-end">
+                            <Button variant="text" disabled classes={{ disabled: classes.viewStatsDisabled }}>
+                                {t('plugin_referral_view_stats')}
+                            </Button>
+                            <Button
+                                disabled
+                                variant="contained"
+                                size="medium"
+                                className={classes.buttonWithdraw}
+                                onClick={() => console.log('request to withdraw')}>
+                                {t('plugin_referral_request_to_withdraw')}
+                            </Button>
+                            <Button
+                                variant="contained"
+                                size="medium"
+                                onClick={() => {
+                                    onAdjustRewardButtonClick({ ...farm })
+                                }}>
+                                {t('plugin_referral_adjust_rewards')}
+                            </Button>
+                        </Box>
+                    }
+                />
+            ))}
+        </>
+    )
+}
+
+export function CreatedFarms(props: PageInterface) {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const { classes: sharedClasses } = useSharedStyles()
+    const currentChainId = useChainId()
+    const requiredChainId = useRequiredChainId(currentChainId)
+    const account = useAccount()
+    const { ERC20 } = useTokenListConstants()
+    const { value: allTokens = [], loading: loadingAllTokens } = useAsync(
+        async () =>
+            !ERC20 || ERC20.length === 0 ? [] : TokenList.fetchERC20TokensFromTokenLists(ERC20, currentChainId),
+        [currentChainId, ERC20?.sort().join()],
```

Comment:
`ERC20` has been memoized, no need to serialize.

```suggestion
        [currentChainId, ERC20],
```